frontend/packages/console-app/src/hooks/useTechPreviewFlagProvider.ts

```diff
@@ -0,0 +1,9 @@
+import { SetFeatureFlag } from '@console/dynamic-plugin-sdk/src/extensions/feature-flags';
+import { FLAG_TECH_PREVIEW } from '../consts';
+
+type UseTechPreviewFlagProvider = (setFeatureFlag: SetFeatureFlag) => void;
+const useTechPreviewFlagProvider: UseTechPreviewFlagProvider = (setFeatureFlag) => {
+  setFeatureFlag(FLAG_TECH_PREVIEW, !!window.SERVER_FLAGS.techPreview);
+};
+
+export default useTechPreviewFlagProvider;
```

Comment:
I personally like named exports (because it means can allow just one name for a given function) but it's okay to have a default on here if you like it more

Reply:
Yeah, I don't have a strong opinion on this.